scalyr_agent/log_processing.py

```diff
@@ -750,6 +738,39 @@ def readline(self, current_time=None):
                     result.timestamp = timestamp
                     if attrs:
                         result.attrs = attrs
+                if self.__merge_json_parsed_lines:
+                    while result.line and not result.line.endswith(b"\n"):
+                        # Docker splits log lines at 16KB, we read lines until we hit a newline character and join them
+                        # so we can enforce our own max line length correctly.
+                        next_line = self.__read_extended_line(current_time)
+                        if len(next_line) == 0:
+                            # If we fail to read a line ending in a newline, we want to seek back to before we started
+                            # building this merged line and hopefully get all the bits next time around. We have a timer
+                            # that uses the line_completion_wait_time configuration used in LineMatchers to output the
+                            # line if we take too long waiting for a newline.
+                            if self.__merge_json_line_time is None:
+                                self.__merge_json_line_time = current_time
+
+                            if (
+                                current_time - self.__merge_json_line_time
+                                < self.__line_completion_wait_time
+                            ):
+                                self.__buffer.seek(original_buffer_position)
+                                self.__position = self.__determine_mark_position(
+                                    self.__buffer.tell()
+                                )
+                                return LogLine(line=b"")
```

Comment:
Should we also log something here (I guess at least at the DEBUG level although since it's an edge case which shouldn't happen often, it may also be OK to use INFO or similar).